classes/search/SearchFileParser.php

```diff
@@ -68,8 +70,10 @@ public function setFilePath($filePath)
      */
     public function open()
     {
-        $this->fp = @fopen($this->filePath, 'rb');
-        return $this->fp ? true : false;
+        if (!($this->fp = @fopen($this->filePath, 'rb'))) {
+            throw new Exception("Failed to parse the file \"{$this->filePath}\"\nLast error: " . error_get_last());
```

Comment:
I think the search parser should be made resilient to missing files; no objection to an exception being thrown here, but if so, then the indexing code should be prepared to catch it and move on with indexing other content.

Reply:
This is supposed to be executed inside a job, so I thought about letting the `Exception` leak, in order to retry the job.

But it's better to index the maximum we can, before failing the job, so I'll recover the previous behavior, but let the `Exception` leak at the end.

Reply:
Ah, regarding your comment, I just the code is already fine, there's a try/catch in the outer scope.
But I'll let an `Exception` leak at the end, in order to retry.